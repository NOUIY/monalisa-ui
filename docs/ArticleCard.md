# ArticleCard

Build awesome article card for your blog articles quickly.

View full example: [ArticleCard Example](https://github.com/tuantvk/monalisa-ui/blob/master/example/ArticleCard/index.js)

```javascript
import React from 'react';
import { View } from 'react-native';
import { ArticleCard } from 'monalisa-ui';

const datas = [
  {
    img: 'http://lorempixel.com/640/480/abstract',
    title: 'Et saepe in aut voluptates illum eum laudantium ipsam similique.',
    category: 'Aut voluptatem dicta',
    author: 'Dr. Ida Marvin',
    time: '21 min ago',
  },
  {
    img: 'http://lorempixel.com/640/480/abstract',
    title: 'Et saepe in aut voluptates illum eum laudantium ipsam similique.',
    category: 'Aut voluptatem dicta',
    author: 'Dr. Ida Marvin',
    time: '21 min ago',
  },
  {
    img: 'http://lorempixel.com/640/480/abstract',
    title: 'Et saepe in aut voluptates illum eum laudantium ipsam similique.',
    category: 'Aut voluptatem dicta',
    author: 'Dr. Ida Marvin',
    time: '21 min ago',
  },
];

const Example = () => (
  <View>
    {
      datas.map((data, index) => (
        <ArticleCard
          key={index}
          {...data}
          style={{ marginBottom: 15 }}
        />
      ))
    }
  </View>
);

export default Example;
```

#### Configuration

| Property      | Default       | Option    | Description  |
| ------------- |:-------------:|:---------:|:------------:|
| img           | -             | -         | The address of the image |
| title         | -             | string    | The title of the article |
| category      | -             | string    | The category of the article |
| author        | -             | string    | The author of the article |
| time          | -             | string    | The time of the article |
| style         | -             | -         | Style for the article card |
| imageStyle    | -             | -         | Style for the image |
| categoryStyle | -             | -         | Style for the category |
| titleStyle    | -             | -         | Style for the title |
| footerStyle   | -             | -         | Style for the footer |
| activeOpacity | -             | number    | Determines what the opacity of the wrapped view should be when touch |