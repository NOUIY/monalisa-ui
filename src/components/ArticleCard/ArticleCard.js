/**
 * @license
 * Copyright tuantvk. All Rights Reserved.
 * Licensed under the MIT License. See License in the project root for license information.
 */

import React, { Component } from 'react';
import {
  View,
  Image,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { Text } from '../Text';
import PropTypes from 'prop-types';
import {
  wScale,
  scale,
} from '../../utils';
import {
  gray,
  blue,
} from '../../styles';
import { withStyle } from '../../theme';

class ArticleCard extends Component {
  render() {
    const {
      img,
      title,
      category,
      author,
      time,
      style,
      imageStyle,
      categoryStyle,
      titleStyle,
      footerStyle,
      styleContent,
      activeOpacity,
      ...rest
    } = this.props;

    return (
      <TouchableOpacity activeOpacity={activeOpacity} {...rest}>
        <View style={[styles.row, style]}>
          <Image
            source={{ uri: img }}
            style={[styles.img, imageStyle]}
          />
          <View style={[styles.content, styleContent]}>
            <View>
              <Text style={[styles.category, categoryStyle]}>{category}</Text>
              <Text style={[styles.title, titleStyle]}>{title}</Text>
            </View>
            <View style={styles.row}>
              <Text style={[styles.footer, footerStyle]}>{author}</Text>
              <Text style={[styles.footer, footerStyle]}>{time}</Text>
            </View>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
  },
  img: {
    width: '35%',
    height: wScale(75),
    marginRight: scale(10),
  },
  content: {
    width: '65%',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  category: {
    color: blue,
  },
  title: {
    fontSize: 16,
  },
  footer: {
    color: gray,
    fontSize: 11,
    marginRight: scale(10),
  }
});

ArticleCard.defaultProps = {
  activeOpacity: 0.9,
}

ArticleCard.propTypes = {
  title: PropTypes.string,
  category: PropTypes.string,
  author: PropTypes.string,
  time: PropTypes.string,
  activeOpacity: PropTypes.number,
}


export default withStyle(ArticleCard);