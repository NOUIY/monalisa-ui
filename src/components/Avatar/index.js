import React from 'react';
import {
  Image,
} from 'react-native';

export default ({
  circle,
  rounded,
  style,
  width = 50,
  height = 50,
  ...rest
}) => (
    <Image
      {...rest}
      style={[
        { width, height },
        circle && { borderRadius: width / 2 },
        rounded && { borderRadius: width / 10 },
        style,
      ]}
    />
  );