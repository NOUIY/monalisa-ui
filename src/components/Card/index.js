import Card from './Card';
import CardImage from './CardImage';

export {
  Card,
  CardImage,
}