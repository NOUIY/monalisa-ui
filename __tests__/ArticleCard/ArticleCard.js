import React from 'react';
import { ArticleCard } from '../../src/components/ArticleCard';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { render } from 'react-native-testing-library';

const datas = {
  img: 'http://lorempixel.com/640/480/abstract',
  title: 'Et saepe in aut voluptates illum eum laudantium ipsam similique.',
  category: 'Aut voluptatem dicta',
  author: 'Dr. Ida Marvin',
  time: '21 min ago',
};

describe('ArticleCard Componnent', () => {

  it('should render without issues', () => {
    const component = shallow(
      <ArticleCard
        {...datas}
      />
    );
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should pass img', () => {
    const component = shallow(
      <ArticleCard img='http://lorempixel.com/640/480/abstract' />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should pass title, category props should still work', () => {
    const { queryByText } = render(
      <ArticleCard title="foo" category="bar" />
    );
    expect(queryByText('foo')).not.toBeNull();
    expect(queryByText('bar')).not.toBeNull();
  });


});