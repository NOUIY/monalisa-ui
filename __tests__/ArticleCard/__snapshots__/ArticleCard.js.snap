// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard Componnent should pass img 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ArticleCard Componnent should render without issues 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
